### PR TITLE
fix new lazy broadcast on 0.7

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

**Comment by SimonDanisch** [comment]
kind of WIP, but does already fix quite a bit ;)

**Comment by andyferris** [comment]
I guess we will actually need to merge this soon to keep up to date with master.

Can we deal with the dynamic stuff after the fact? @tkoolen?

**Comment by SimonDanisch** [comment]
Are there tests for the dynamic stuff? I got a new fix and now pass most broadcast tests...

**Comment by tkoolen** [comment]
Thanks for the PR. I'm working on some changes. I'll submit a PR against your PR branch in a bit.

**Comment by tkoolen** [comment]
See https://github.com/JuliaArrays/StaticArrays.jl/pull/408.

File: src/broadcast.jl [hunk 1]
```diff
@@ -46,7 +46,7 @@
 else
     ## New Broadcast API ##
     import Base.Broadcast:
-    BroadcastStyle, AbstractArrayStyle, broadcast
```
Comment (tkoolen) [inline src/broadcast.jl]: Ah, I guess the `materialize!` import I added is unnecessary.

**Comment by SimonDanisch** [comment]
fixed by #408

**Comment by tkoolen** [comment]
No! That PR is against the sd/lazybcast branch!

**Comment by SimonDanisch** [comment]
Woops, of course ;) Thanks!

**Pull request merged**